docs: Update the information for remotecfg, import.http, and add new topic about loading remote config files (#5758)
grafana/alloy: docs/sources/configure/load-remote-configuration.md

Modules define reusable components using `declare` blocks, which the local configuration then instantiates.
After you import a module, you must instantiate its declared components in the local configuration for them to run.

{{< admonition type="note" >}}
You can't point {{< param "PRODUCT_NAME" >}} directly at a remote URL on startup.
You must have a local configuration file that uses `import.http` to import modules from the remote server.
The remote file must define reusable components using `declare` blocks.
{{< /admonition >}}

{{< param "PRODUCT_NAME" >}} periodically polls the URL to detect and apply configuration changes.

### Module requirements

Modules must define reusable components using `declare` blocks and can't contain top-level configuration blocks such as `logging` or `remotecfg`.
You configure CLI flags when starting {{< param "PRODUCT_NAME" >}}, not within modules.

The following module is invalid because it contains a `logging` block, which isn't allowed in modules:

```alloy
logging {
  level = "debug"
}

declare "pipeline" {
  prometheus.scrape "default" {
    targets = [{"__address__" = "localhost:9090"}]
  }
}
```

> The following module is valid because it only contains `declare` blocks:

Reviewer: ```suggestion
The following module is **valid** because it only contains `declare` blocks, which is the simplest module pattern:
```

```alloy
declare "pipeline" {
  prometheus.scrape "default" {
    targets = [{"__address__" = "localhost:9090"}]
  }
}
```

Global configuration such as `logging` must remain in the local configuration file that imports the module.

### Create the remote configuration file

Create a configuration file on your HTTP server.

> The file must be a valid {{< param "PRODUCT_NAME" >}} module, using top-level `declare` blocks to define the reusable configuration you’ll import.

Reviewer: ```suggestion
The file must be a valid {{< param "PRODUCT_NAME" >}} module, using top-level `declare` blocks to define reusable configuration that is imported and instantiated locally.
```

Use `import.http` to load configuration modules from an HTTP server.
This is the recommended approach when you have configuration modules hosted on a web server.

{{< param "PRODUCT_NAME" >}} periodically makes HTTP requests to the configured URL and treats the response body as {{< param "PRODUCT_NAME" >}} module content.
The response doesn't need a special API.

> A standard HTTP server that returns {{< param "PRODUCT_NAME" >}} text is enough.

Reviewer: Tiny suggestion maybe to just smush these two sentences together?

{{< param "PRODUCT_NAME" >}} loads remote content with `import.http` and treats it as a module, not as a complete top-level configuration file.
Modules define reusable components in `declare` blocks.

> After you import a module, you must instantiate its declared components in the local configuration so they run.

Reviewer: This tripped me up because I immediately thought "omg how do i instantiate a component??", but does this literally just mean like... after import, for the component to actually get used, you have to "use" it in your local config pipelines?

Author: Tweaked it a little to be less wordy and more direct.
